src/Tools/dotnet-watch/BrowserRefresh/test/WebSockerScriptInjectionTest.cs

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Watch.BrowserRefresh
+{
+    public class WebSockerScriptInjectionTest
+    {
+        [Fact]
+        public async Task TryInjectLiveReloadScriptAsync_DoesNotInjectMarkup_IfInputDoesNotContainBodyTag()
+        {
+            // Arrange
+            var stream = new MemoryStream();
+            var input = Encoding.UTF8.GetBytes("<div>this is not a real body tag.</div>");
+
+            // Act
+            var result = await WebSocketScriptInjection.TryInjectLiveReloadScriptAsync(input, 0, input.Length, stream);
```

Comment:
This needs tests that vary the offset and count, there are several bugs in the current implementation.

Reply:
I think I made some last minute changes, that were obviously incorrect. Let me change this back